frontend/resourceadm/pages/DeployResourcePage/DeployResourcePage.tsx

```diff
@@ -29,75 +27,44 @@ export const DeployResourcePage = ({ navigateToPageWithError }: Props) => {
   const { selectedContext, resourceId } = useParams();
   const repo = `${selectedContext}-resources`;
 
-  // TODO - Tanstack: https://tanstack.com/query/latest
-  const [loadingPublishStatus, setLoadingPublishStatus] = useState(false);
-  const [loadingValidatePolicy, setLoadingValidatePolicy] = useState(false);
-  const [loadingValidateResource, setLoadingValidateResource] = useState(false);
-  const [hasError, setHasError] = useState(false);
-
   const [isLocalRepoInSync, setIsLocalRepoInSync] = useState(false);
-
-  const [hasResourceError, setHasResourceError] = useState(true);
   const [hasPolicyError, setHasPolicyError] = useState<'none' | 'validationFailed' | 'notExisting'>(
     'none'
   );
+  const [newVersionText, setNewVersionText] = useState('');
 
+  // Queries to get metadata
   const { data: repoStatus } = useRepoStatusQuery(selectedContext, repo);
-
-  const [newVersionText, setNewVersionText] = useState('');
-  const [versionInTest, setVersionInTest] = useState('');
-  const [versionInProd, setVersionInProd] = useState('');
-  const [localVersion, setLocalVersion] = useState('');
+  const { data: versionData, isLoading: versionLoading } = useResourcePolicyPublishStatusQuery(
+    selectedContext,
+    repo,
+    resourceId
+  );
+  const { data: validatePolicyData, isLoading: validatePolicyLoading } = useValidatePolicyQuery(
+    selectedContext,
+    repo,
+    resourceId
+  );
+  const { data: validateResourceData, isLoading: validateResourceLoading } =
```

Comment:
I agree that on a "normal" API call that would be the best. The issue here is that even though the API call is success, the validation might give a 400 because the status code is added to the object in backend. 
This is the object returned in the data from onSuccess: 
<img width="543" alt="Screenshot 2023-07-31 at 07 45 22" src="https://github.com/Altinn/altinn-studio/assets/133344438/3cf40325-c0ab-4a5c-8e8e-d5935c2b4926">

For the policy, however, I can use it for one of the checks because if the API call gives 404 it means that the policy is yet not created.